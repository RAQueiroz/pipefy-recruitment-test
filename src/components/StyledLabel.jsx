import styled from "styled-components";

const StyledLabel = styled.label`
  display: ${props => (props.inline ? "block" : "grid")};
  font-weight: 600;
  line-height: 40px;
  color: #333;
  padding: 10px;
`;

export default StyledLabel;
