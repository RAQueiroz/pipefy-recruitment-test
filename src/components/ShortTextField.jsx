import React from "react";
import styled from "styled-components";
import StyledLabel from "./StyledLabel";
import Colors from "../utils/Colors";
import PropTypes from "prop-types";

const StyledInput = styled.input`
  border: solid 1px ${props => (props.error ? Colors.errorLight : "#ccc")};
  line-height: 40px;
  padding-left: 10px;
  border-radius: 5px;
`;

const ShortTextField = ({
  label,
  id,
  error,
  value = "",
  onChange = () => {}
}) => (
  <div>
    <StyledLabel>
      {label}
      <StyledInput
        error={error}
        value={value}
        onChange={({ target }) => onChange({ id, value: target.value })}
        type="text"
        id={id}
      />
    </StyledLabel>
  </div>
);

ShortTextField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func
};
export default ShortTextField;
