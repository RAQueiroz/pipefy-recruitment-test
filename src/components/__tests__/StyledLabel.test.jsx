import React from "react";
import ShortTextField from "../ShortTextField";
import StyledLabel from "../StyledLabel";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(" <StyledLabel/>", () => {
  const label = shallow(<StyledLabel>Name</StyledLabel>);

  it("renders", () => {
    expect(label.containsMatchingElement("Name")).toBeTruthy();
  });

  it("to match snapshot", () => {
    expect(toJson(label)).toMatchSnapshot();
  });
});
