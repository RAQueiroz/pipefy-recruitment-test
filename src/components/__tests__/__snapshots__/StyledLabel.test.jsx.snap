// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <StyledLabel/> to match snapshot 1`] = `
<label
  className="sc-bdVaJa hOggSJ"
>
  Name
</label>
`;
